soda_machine.rb

```diff
@@ -1,18 +1,41 @@
 class SodaMachine
   attr_reader :sodas, :cash
 
+  @@inventory_count = 0
   def initialize(args = {})
     @sodas = args[:sodas]
     @cash = args[:cash]
+    @@inventory_count += :sodas.length - 1
   end
 
   def current_inventory_count
+    @@inventory_count
   end
 
   def find_soda(soda_brand)
+    @sodas.each do |soda|
+      if soda.brand == soda_brand
+        return soda
+      else
+        return nil
+      end
+    end
   end
 
+
   def sell(soda_brand)
+    @sodas.keep_if { |soda| soda == soda_brand}
+    @sodas.each do |soda|
+      if soda.brand == soda_brand
+        SodaMachine.cash += soda.price
+      else
+        return nil
+      end
+    end
+    # end
+    # @cash += soda.cash
+      
   end
```

Comment:
@martymclaugh there a few things.
Think what is are the returning values out of Array#keep_if, how you can use that later.
then what's SodaMachine.cash if you have `@cash`
The are you gonna return in the else on any element of the iteration?

Here some alternatives:

```ruby
  def find_soda(soda_brand)
    @sodas.find {|soda| soda.brand == soda_brand}
  end

  def sell(soda_brand)

    # Alternative 1

    # if find does not find returns nil
    item = @sodas.find {|soda| soda.brand == soda_brand}

    # nil evaluates as false (nil ? true : false  # => false )
    if item
      @sodas.delete(item)
      @cash += item.price
    end

    item

    # ================

    # Alternative 2

    # Select an array
    item = @sodas.select {|soda| soda.brand == soda_brand}

    if item.length > 0
      @sodas.delete(item[0])
      @cash += item[0].price
      return item[0]
    end

    nil

    # ================

    # Alternative 3

    index = @sodas.index {|soda| soda.brand == soda_brand}
    item = nil

    if index 
      item = @sodas[index]
      @cash += item.price
      @sodas.delete_at(index)
    end
    
    item

    # ================

    # Alternative 4

    item = nil

    @sodas.each_with_index do |soda, index|
      if soda.brand == soda_brand
        item = soda
        @cash += soda.price
        @sodas.delete_at(index)
      end
    end

    item
  end